test/api/route/messages_test.dart

```diff
@@ -235,7 +235,7 @@ void main() {
 
       connection.zulipFeatureLevel = 270;
       checkNarrow(eg.topicNarrow(12, 'stuff', with_: 1).apiEncode(), jsonEncode([
-        {'operator': 'stream', 'operand': 12},
+        {'operator': 'channel', 'operand': 12},
```

Comment:
This test change shows that the effect of this commit is a bit broader than what the commit message says:

    internal_link: Generate new narrow links with "channel", not "stream"

It's also changing the get-messages requests we make to the server.

I think it's fine for that to happen in the same commit; given the way ApiNarrow works, I think splitting those changes into separate commits would require complicating the logic. But the commit message should be adjusted to make that clear.